lib/Dialect/EmitC/Conversion/TosaToEmitC.cpp

```diff
@@ -117,6 +117,36 @@ class FullyConnectedOpConversion
   }
 };
 
+class MatMulOpConversion : public OpConversionPattern<tosa::MatMulOp> {
+  using OpConversionPattern<tosa::MatMulOp>::OpConversionPattern;
+
+public:
+  MatMulOpConversion(MLIRContext *ctx, StringRef funcName)
+      : OpConversionPattern<tosa::MatMulOp>(ctx) {}
+
+private:
+  LogicalResult
+  matchAndRewrite(tosa::MatMulOp matMulOp, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (matMulOp.quantization_info().hasValue()) {
+      return matMulOp.emitError(
+          "Quantization of tosa.matmul is currently not supported.");
+    }
+
+    StringRef funcName = "tosa::matmul";
+    StringAttr callee = rewriter.getStringAttr(funcName);
+
+    Type type = matMulOp.getType();
+
+    ArrayAttr args;
+    ArrayAttr templateArgs;
+
+    rewriter.replaceOpWithNewOp<emitc::CallOp>(matMulOp, type, callee, args,
```

Comment:
I suggest to directly pass the type here
```suggestion
    rewriter.replaceOpWithNewOp<emitc::CallOp>(matMulOp, matMulOp.getType(), callee, args,
```
and drop it above. That's the way we do it in most conversions.